### ISSUE TITLE
GOGAPPI - compliance check

### ISSUE THREAD

**Issue by BLR-0118** [ONDC-Official/verification-logs#304]
1. /search - ttl should be 30s as per API contract (this is needed to block replay attacks);
2. /select - 
a. what's the purpose of putting all these extra info here e.g. billing comes in /init, not /select; Also, don't need to be backward compatible with "fulfillment";
b. How's the timestamp for /select earlier than /search?
c. what does items.location_id of "None" mean? (this is not the same as location_id in /on_search);
3.  /init - suggest you remove the payment block (here it's proposed as COD but in /confirm, it becomes prepaid);
4. From /select to /confirm, the transaction_id and message_id are same. Transaction_id should be the same after /search for all other APIs. Message_id is used to identify request/response pair and new id should be created for each API request;

**Comment by BLR-0118** [comment]
@ShishirPrashanth

**Comment by ShishirPrashanth** [comment]
All issues resolved in https://github.com/ONDC-Official/verification-logs/pull/311

1. done
2a. done
2b. done (we cache catalog so we had finished all the tests and later just done a search pull, fixed in the updated pr)
2c. fixed
3. done
4. done

**Comment by ShishirPrashanth** [comment]
4 done

**Comment by BLR-0118** [comment]
1. /select - fulfillments.end.location & contact come in /init, not in /select as per API contract. Pls make sure the fulfillments & items struct are exactly as per API contract [here](https://docs.google.com/document/d/12oMC4A05jT5rjCbY-M7j4T2oekr9bqcvSOpTI0jFEgE/edit#)
2. /init - 
a. quote.price doesn't add up to prices in breakup;
b. items.location_id should be match one of the values of locations.id;
3. /confirm - 
a. no payment.params.transaction_status in API contract;
b. "@ondc/org/buyer_app_finder_fee_type":"precent" is invalid;
4. No APIs for /update, /support, /track (generally provided by all F&B apps)

**Comment by BLR-0118** [comment]
@ShishirPrashanth

**Comment by ShishirPrashanth** [comment]
1. fulfillments.end.location is part of select as per the contract spec. contact is only part of init but since we handle both inputs together I send it in both. I don't see any harm in sending an extra key since it won't be consumed anyways.
2. b. items.location_id is compulsory only during select and items.fulfillment_id is compulsory only during init. However we have the same popo that need to handle it so during select item.fulfillment_id will be sent as null and during init item.location_id will be sent as null. Correct values are sent when needed.

3. a & b - minor errors, fixed in backend, will resubmit new logs with the fix.
2. a - Need to look into this thoroughly. Will resubmit with the fix.

**Comment by ShishirPrashanth** [comment]
4. /update is not supportedfor gogappi since it is only in F&B so returns and cancels are generally not allowed. /track is not yet supported but will be supported after the initial release.

Will enable /support and update logs for that as well.

**Comment by BLR-0118** [comment]
2b. items.fulfillment_id is assigned by seller app in on_select & carried forward until confirm. In case of multiple fulfillments, this is the only way to know which items are in which fulfillment.
items.location_id is provided in /select and should be carried forward wherever it is required as per the API contract. Again, this has to be one of the ids from on_search. This helps to identify, for a store with multiple locations, which location is being used.
Suggest you make the changes as requested.

**Comment by BLR-0118** [comment]
@ShishirPrashanth

**Comment by ShishirPrashanth** [comment]
Yes @BLR-0118 we are doing exactly that. Since item.fulfillment_id comes during on_select, it is set to null during select call. It carries forward till confirm.
In case of item.location_id as well it is one of the store locations from on_search. This is only necessary in select so we send it then. During init it is not part of the contract so it becomes null and doesn't carry forward. As per the contract it doesn't need to be carried forward anywhere.

**Comment by ShishirPrashanth** [comment]
1, 2b are not issues, they are working as expected
2a, 3ab were minor issues that have been fixed, I have collected logs with the fixes.
4 - support, on_support logs added. update is not supported since gogappi is only in F&B. Track is not implemented yet, will be taken up after initial launch.

@BLR-0118 please confirm so that I resubmit logs with the fixes as mentioned.

**Comment by BLR-0118** [comment]
Logs with Uengage:
1. /select - pls make sure you provide info exactly as per API contract;
2. /init - why is items.location_id null here? Also, keys such as max/available qty, quote shouldn't be passed here;
3. /confirm - quote.price.value should be string;
4. No /track API - this is normally provided by F&B apps;
Pls make sure compliance with API contract is "to the tee".

**Comment by BLR-0118** [comment]
@ShishirPrashanth

**Comment by BLR-0118** [comment]
@arvind9642

**Comment by BLR-0118** [comment]
Pls also ensure you comply with [this](https://github.com/ONDC-Official/verification-logs/blob/main/Tech%20Compliance%20Checklist%20for%20Prod.md)

**Comment by ShishirPrashanth** [comment]
Fixed in https://github.com/ONDC-Official/verification-logs/pull/327 :

1. Done - Now only necessary keys are sent.
2. Done - removed items.location_id (not compulsory in contract spec). Also removed quote.
3. Done
4. /track is not yet implemented. This will be done in the future.

All logs updated for Uengage.
Received NACK during on_confirm with Sellerapp. In real life /cancel and /status wouldn't be called but force called those for the sake of log submission. This also works out to be a good example for /support API when order is not placed.

@BLR-0118 
CC: @arvind9642

**Comment by BLR-0118** [comment]
Logs with uengage:
1. /select - why sending contact phone/email in fulfillments?
2. /init, /confirm - remove fulfillments.tracking since that's supposed to come from the logistics provider;
3. /confirm - don't include null key for item_quantity for tax, delivery;

**Comment by BLR-0118** [comment]
4. Why don't you enable /track now?
5. No /update API?

@arvind9642 @ShishirPrashanth - pls fix the above & resubmit only these parts. We should also schedule an app walkthru later this week.

Also, check your compliance with [this](https://github.com/ONDC-Official/verification-logs/blob/main/Tech%20Compliance%20Checklist%20for%20Prod.md)

**Comment by ShishirPrashanth** [comment]
Fixed in https://github.com/ONDC-Official/verification-logs/pull/333:

1. Fixed - not sending anymore
2. Done - removed
3. Done - removed
4. Done - enabled /track, added logs with Uengage as well as Sellerapp. 
5. /update not supported since we will only be in F&B

@BLR-0118 resubmitted select, init, confirm. Added track.
CC: @arvind9642
